### ISSUE TITLE
internal consistency problem when checking if paper.bbl changed

### ISSUE THREAD

**Issue by KonradHoeffner** [tectonic-typesetting/tectonic#1150]
tectonic reruns TeX compilation multiple times until it hits the limit of 6 passes, but I have no idea why.
This warning seems to be emitted by <https://github.com/tectonic-typesetting/tectonic/blob/fed29a8339e09b733a423ccc20bb440363dbc5bc/src/driver.rs#L1364>.

```
/tmp$ git clone git@github.com:annosaxfdm/anno-paper-swj paper
/tmp$ cd paper
paper$ git checkout 1af65946b752e453494f4f806fc7fb7b1f73deb4
paper$ tectonic paper.tex                                                                                                                        heads/master
Running TeX ...
Running BibTeX on paper.aux ...
Rerunning TeX because bibtex was run ...
Rerunning TeX because "paper.out" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
warning: TeX rerun seems needed, but stopping at 6 passes
Running xdvipdfmx ...
warning: ICC profile format spec. version 4.4.0 not supported in current PDF version setting.
warning: ICC profile not embedded.
```

**Comment by derchr** [comment]
Probably it's the same issue as [1146](https://github.com/tectonic-typesetting/tectonic/issues/1146).

**Comment by KonradHoeffner** [comment]
Oh sorry, I didn't found that one because it doesn't contain "warning: internal consistency problem", however could that mean that this is a different issue or are warnings just deactivated in the other issue output?

**Comment by CraftSpider** [comment]
If anyone experiencing the issue is willing to try #1155, that should fix the issue by making bibtex correctly close log files on success.

**Comment by KonradHoeffner** [comment]
@CraftSpider: I tried your bibtex-fix branch and while `cargo vcpkg` reported a failure (see below) I think that is unrelated to the branch, it still built successfully and the issue is indeed fixed, thank you!

## cargo vcpkg build error  (probably unrelated, still builds)

```
cargo vcpkg build                                                                                                                        bibtex-fix
     Cloning https://github.com/microsoft/vcpkg
    Checkout rev 4a600e9fea71bd7872080cbb716797e04d30e6d3
   Compiling vcpkg
  Installing fontconfig freetype harfbuzz[graphite2] icu
-- stdout --

-- stderr --

cargo-vcpkg: failed
```
## tectonic from the Arch Linux package

```
paper$ tectonic paper.tex
Running TeX ...
Running BibTeX on paper.aux ...
Rerunning TeX because bibtex was run ...
Rerunning TeX because "paper.aux" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
Rerunning TeX because "paper.bbl" changed ...
warning: internal consistency problem when checking if paper.bbl changed
warning: TeX rerun seems needed, but stopping at 6 passes
Running xdvipdfmx ...
warning: ICC profile format spec. version 4.4.0 not supported in current PDF version setting.
warning: ICC profile not embedded.
Writing `paper.pdf` (2.29 MiB)
Skipped writing 3 intermediate files (use --keep-intermediates to keep them)
```

## tectonic CraftSpider fork bibtex-fix branch (issue is fixed)

```
paper$ ~/tmp/tectonic/target/debug/tectonic --version
tectonic 0.0.0-dev.0Tectonic 0.0.0-dev.0
paper$ ~/tmp/tectonic/target/debug/tectonic paper.tex
Running TeX ...
Running BibTeX on paper.aux ...
Rerunning TeX because bibtex was run ...
Rerunning TeX because "paper.aux" changed ...
Running xdvipdfmx ...
warning: ICC profile format spec. version 4.4.0 not supported in current PDF version setting.
warning: ICC profile not embedded.
Writing `paper.pdf` (2.29 MiB)
Skipped writing 3 intermediat
```

**Comment by CraftSpider** [comment]
Should be fixed by #1155, please re-open if that doesn't fix this once released.
